lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -263,3 +246,60 @@ function dosomething_rogue_rb_exists_in_rogue($rbid)
 {
   return db_query("SELECT * FROM {dosomething_rogue_reportbacks} rogue_rbs WHERE rbid = :rbid", array(':rbid' => $rbid))->fetchAll();
 }
+
+/**
+  * Transform reportback into the appropriate scheme to send to Rogue.
+  *
+  * @param $values - Reportback values to send to Rogue.
+  * @param $northstar_id - User's Northstar user id.
+  * @return array
+  */
+ function dosomething_rogue_transform_reportback($values, $northstar_id) {
+   if ($values['nid']) {
+     $campaign_id = $values['nid'];
+   }
+   elseif ($values['campaign_id']) {
+     $campaign_id = $values['campaign_id'];
+   }
+
+   $run = dosomething_helpers_get_current_campaign_run_for_user($campaign_id);
+
+   $data = [
+     'northstar_id' => $northstar_id,
+     'campaign_id' => $campaign_id,
+     'campaign_run_id' => $run->nid,
+     'quantity' => $values['quantity'],
+     'why_participated' => $values['why_participated'],
+     'file' => isset($values['file']) ? $values['file'] : NULL,
+     'caption' => isset($values['caption']) ? $values['caption'] : NULL,
+     'status' => isset($values['status']) ? $values['status'] : 'pending',
+     'source' => isset($values['source']) ? $values['source'] : NULL,
+     'type' => isset($values['type']) ? $values['type'] : NULL,
+   ];
+
+   if ($values['crop_x']) {
```

Comment:
I created a function once that just extracted properties from an array. It was a little cleaner. So you can do like 

```
/**
 * Given an array of key/value pairs, create a new array with only the
 * keys given.
 *
 * @param  associative array  $data
 * @param  array $desiredKeys
 *
 * @return array
 */
function extract_values_by_keys($array, $desiredKeys)
{
    $desiredValues = array_intersect_key($array, array_flip($desiredKeys));
    return (count($array) > 0) ? $desiredValues : null;
}
```

and then use it like:

```
$cropped_values = extract_values_by_keys($values,['crop_x', 'crop_y'....]); //gives you an array

$data = array_merge($data, $cropped_values);
```

allows you to remove some of the duplicated lines and the extra loop at the end.